mono_repo/lib/src/basic_config.dart

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'coverage_processor.dart';
+
+abstract class BasicConfiguration {
+  Set<CoverageProcessor> get coverageProcessors;
```

Comment:
Trying to minimize the amount of random state we sent to individual tasks. Not strictly required. Happy to discuss.